### DISCUSSION TITLE
Where can I find release notes for an individual `Microsoft.Extensions.*` library?

### DISCUSSION THREAD

**Discussion by julealgon** [dotnet/runtime#103345] [Q&A]
I'm in the process of simplifying some of our dependencies here in a large solution that was migrated from legacy `csproj` to the new SDK-style `csproj` recently.

I came across a case where we have the following library version that I was contemplating downgrading for simplicity:
- `Microsoft.Extensions.Options` `v8.0.2`

I wanted to see what the changes were between `v8.0.0` and `v8.0.2` to see whether downgrading would be acceptable but I couldn't find this information anywhere.

I started looking here in the `.NET 8.0.2` release:
- https://github.com/dotnet/runtime/releases/tag/v8.0.2

But none of the links or references seem to point to actual specific changes in that library, it is only listed as participating in the release.

I also didn't find any `CHANGELOG.md` document in the library folder itself, which is another common place to store changes on a per-project basis.

Where can I find release notes for `Microsoft.Extensions.Options` and other `Microsoft.Extensions.*` libraries? Are they actually available somewhere? If not, what is the best alternative method to check the differences between versions?

**Comment by huoyaoyuan** [comment]
The patch version of individual package isn't the same of shared framework. If there's no update to a package, its version won't be changed.

To get the corresponding shared framework release (source tag), the simplest way is to inspect the release date. For example Extensions.Options 8.0.2 was released at Feb 13, so it belongs to 8.0.2 tag at Feb 13. It's just a coincidence for the numbers to be same.

> If not, what is the best alternative method to check the differences between versions?

You can check the git history of the library *from the corresponding tag*.

**Reply by julealgon** [reply]
That's...... fairly convoluted. Have the team considered adding a proper `CHANGELOG.md` file to each project and documenting the changes "properly"?

**Reply by julealgon** [reply]
Also... how do I do this?
> You can check the git history of the library from the corresponding tag.

I don't see any way to run a compare only for a specific folder in github, only for the entire repo. How can I compare only changes from v8.0.0 to v8.0.2 for `Microsoft.Extensions.Options` _only_?

**Reply by huoyaoyuan** [reply]
I don't use compare for this case, but history only. The history of one folder won't typically be long in release branch.

**Reply by julealgon** [reply]
> I don't use compare for this case, but history only. The history of one folder won't typically be long in release branch.

Ah I see what you mean now, thanks @huoyaoyuan .
